dlf/dlflistfacility.man
.lf 1 dlflistfacility.man
.\" @(#)$RCSfile: dlflistfacility.man,v $ $Revision: 1.1 $ $Date: 2003/08/20 13:04:09 $ CERN IT-ADC Vitaly Motyakov
.\" Copyright (C) 2003 by CERN/IT/ADC
.\" All rights reserved
.\"
.TH DLFLISTFACILITY 1 "$Date: 2003/08/20 13:04:09 $" CASTOR "DLF Administrator Commands"
.SH NAME
dlflistfacility \- list all existing facilities in the CASTOR
DLF database
.SH SYNOPSIS
.B dlflistfacility
..SH DESCRIPTION
.B dlflistfacility
lists all existing facilities in the CASTOR DLF database in tabular form:
.HP 1.2i
facility_number		facility assigned number
.HP
facility_name		name of the facility
.SH EXAMPLE
.nf
.ft CW
dlflistfacility
List of facilities:
===================
160     test3
200     testx
201     test1
202     test2
255     dlftest
.ft
.fi
.SH EXIT STATUS
This program returns 0 if the operation was successful or >0 if the operation
failed.
.SH SEE ALSO
.BR Cupvlist(1) ,
.SH AUTHOR
\fBCASTOR\fP Team <castor.support@cern.ch>


CHANGELOG.md
## 5.11.23.0-1 (2026-09-25)

### New Features

- [frontend] Add user configuration improvements for cta-admin-grpc (cta/CTA!1635)

## 5.11.22.0-1 (2026-09-04)

### New Features

- [frontend] Implement JWT blacklisting on the CTA Frontend (cta/CTA!1548)

- [runtime] Add additional validation logic to application config (cta/CTA!1567)


### Bug Fixes

- [scheduler] Fix issues after pg1 release rest: repack admin commands, archive...

- [Tools] Fix 'priority' column in 'cta-admin dr ls' command (cta/CTA!1546)

- [runtime] Make signals, telemetry and health handling more robust (cta/CTA!1554)

- [runtime] Add bounds checking and better error diagnostics to TOML loading (cta/CTA!1559)

## 5.11.21.0-1 (2026-08-25)

### New Features

- [scheduler] Add scheduler workflow metrics to OpenTelementry (cta/CTA!1387)


### Changes

- [Tools] Move cta-admin af ch command to tf ch (cta/CTA!1525)
- [Tools] Replace cta-objectstore-tools.conf with cta-scheduler.conf (cta/CTA!1540)
- [maintd] Change scheduler.objectstore_backend_path to scheduler.config_file (cta/CTA!1518)


### Bug Fixes

- [frontend] Fix logic bug in frontend authentication and SEGFAULT in cta-admin (cta/CTA!1558)
- [Tools] Fix cta-verify-file (cta/CTA!1545)
- [Tools] Add `--storage-class` option to `cta-send-event` (cta/CTA!1542)


### Performance Improvements

- [taped] Skip block reposition request if drive is already positioned correctly (cta/CTA!1527)

## 5.11.20.0-1

### New Features

- [Tools] Add optional storage class statistics, per tape, to cta-admin tool (cta/CTA!1410)
- [Misc] Add event_name to important CTA log messages (cta/CTA!1381)

### Changes

- [Tools] Change short option for cta-admin af ch --idfile from -I to -F (cta/CTA!1453)
- [taped] Taped no longer sets its own user, group and SYS_RAWIO capability (cta/CTA!1426)
- [Misc] Improve .service and .logrotate configuration (cta/CTA!1524)
- [Misc] Modernise RPM spec metadata and file handling (cta/CTA!1507)
- [Misc] Move systemd units to `/usr/lib/systemd/system` (cta/CTA!1497)
- [Misc] Change systemd to weak RPM dependency of CTA (cta/CTA!1509)
- [Misc] Change logrotate to weak RPM dependency of CTA (cta/CTA!1505)
- [Misc] Remove RPM changelog; package CHANGELOG.md as doc (cta/CTA!1508)
- [Misc] Remove noreplace from example configuration files (cta/CTA!1506)
- [taped] Drive process crashing produces CRIT message instead of INFO (cta/CTA!1499)
- [taped] cta-eosdf no longer sources sysconfig (cta/CTA!1455)

### Bug Fixes

- [frontend] Forbid reclaiming tapes with recently deleted files (cta/CTA!1418)
- [frontend] PREPARE gRPC request shouldn't fail without a storage class (cta/CTA!1471)
- [taped] Fix crash due to accidental pointer arithmetic in exception msg (cta/CTA!1502)
- [taped] Fix opentelemetry shutdown race condition with objectstore (cta/CTA!1500)
- [scheduler]: Fix archive unmount decision logic for postgres scheduler (cta/CTA!1450)
- [frontend] Fix frontend catalogue cache for mount rules (cta/CTA!1490)
- [CI] Ensure CTA images trust CERN Root CA (cta/CTA!1535)
- [Misc] Ensure CMake correctly installs executables as PROGRAMS (cta/CTA!1523)

### Security

- [CI] Generate SBOM for each Docker image (cta/CTA!1532)

### Other

- [CI] Add system tests to cover staging tokens on EOS-CTA (cta/CTA!1493)

## 5.11.19.0-1

### Changes

- [frontend] Split Admin/WFE frontends, add mTLS (cta/CTA!1367)
- [catalogue] Forbid adding or changing encryption key of non-empty tapes (cta/CTA!1414)

### Deprecations

- [frontend] Deprecate storage class field in EOS-CTA xattrs (cta/CTA!1424)

### Other

- [CI] Add Enstore and EnstoreLarge tape tests (cta/CTA!1412)

## 5.11.18.0-2

### Bug Fixes

- [Misc] Update required runtime version of opentelemetry-cpp to v1.26.0 (cta/CTA!1370)

## 5.11.18.0-1

### New Features

- [maintd] Add catalogue and telemetry conf files to runtime directory (cta/CTA!1303)
- [Misc] Package basic JSON logging schema (cta/CTA!1304)
- [Misc] Add source location and CTA version to log lines (cta/CTA!1348)

### Changes

- [frontend] Allow admin commands or physics workflows to be disabled on CTA frontend (cta/CTA!1336)
- [Misc] Replace reflect-cpp library with custom implementation (cta/CTA!1291)
- [Misc] Change Catalogue backend install behaviour (cta/CTA!1295)

### Bug Fixes

- [catalogue] Improve CTA resilience to catalogue DB outages during data fetching (cta/CTA!1334)
- [taped] Improve readExactBlock error message contents (cta/CTA!1311)
- [taped] Enabling reading of tapes with misformed OSM tape format (cta/CTA!1343)
- [frontend] Prevent 'cta-admin dr rm' command from crashing the cta-frontend (cta/CTA!1305)
- [frontend] Remove race conditions in NegotiationRequestHandler (cta/CTA!1310)
- [frontend] Fix broken cta-admin fr ls command (cta/CTA!1340)
- [Misc] Fix compile issues when building CTA in Release mode (cta/CTA!1207)

### Performance Improvements

- [CI] Use python xrootd persistent clients for archive instead of xrdcp --parallel (cta/CTA!1274)

### Other

- [Misc] Update opentelemetry-cpp to v1.26.0 (cta/CTA!1321)


## 5.11.17.0-1

### New Features

- [maintd] Add CLI option to use runtime directory for app metadata (cta/CTA!1266)

### Changes

- [maintd] Update maintd to use TOML configuration (cta/CTA!1246)
- [maintd] Use declarative config to initialise OpenTelemetry SDK (cta/CTA!1248)
- [maintd] Add use_unix_domain_socket configuration entry for health server (cta/CTA!1282)
- [maintd] Remove daemon_user and daemon_group settings (cta/CTA!1252)

### Bug Fixes

- [frontend] Fix crash in 'cta-admin dr ls' command when using multiple...
- [frontend] Add error message when trying to change media type of non-empty tape (cta/CTA!1253)

### Removals

- [frontend] Stop using TAPE_POOL.SUPPLY column (cta/CTA!1245)

### Performance Improvements

- [taped] Optimise DriveState queries (cta/CTA!1198)

### Other

- [Misc] Rename *.conf.example files to *.example.conf (cta/CTA!1279)

## 5.11.16.1-1

### Bug Fixes


dlf/dlflistfacility.c
/*
 * 
 * Copyright (C) 2003 by CERN/IT/ADC
 * All rights reserved
 *
 */

#ifndef lint
static char sccsid[] = "@(#)$RCSfile: dlflistfacility.c,v $ $Revision: 1.1 $ $Date: 2003/08/20 13:03:49 $ CERN IT-ADC/CA Vitaly Motyakov";
#endif /* not lint */

#include <errno.h>
#include <grp.h>
#include <pwd.h>
#include <stdio.h>
#include <sys/types.h>
#if defined(_WIN32)
#include <winsock2.h>
#endif
#include "marshall.h"
#include "serrno.h"
#include "dlf.h"
#include "dlf_api.h"

main(argc, argv)
int argc;
char **argv;
{
	U_BYTE fac_no;
	dlf_msg_text_slist_t text_list;
	dlf_msg_text_t* p;
	char fac_name[DLF_MAXFACNAMELEN + 1];

	int c;
	gid_t gid;
	int msglen;
	char *q;
	char *sbp;
	char sendbuf[DLF_REQBUFSZ];
	uid_t uid;
	int status;
	int flags;
	char* reply;
	int rep_size;
	char* rep_buf_end;
	char* rbp;
	int rep_type;
	int socket;
	dlf_log_dst_t* dst;
	int end_rc;

#if defined(_WIN32)
	WSADATA wsadata;
#endif
 
#if defined(_WIN32)
	if (WSAStartup (MAKEWORD (2, 0), &wsadata)) {
		fprintf (stderr, DLF52);
		exit (SYERR);
	}
#endif
	if (dlf_init("DLF-CONTROL") < 0 ) {
	    fprintf (stderr, "Error in initializing global structure.\n");
	    exit (SYERR);
	}

	uid = geteuid();
	gid = getegid();


	/* Build request header */

	sbp = sendbuf;
	marshall_LONG (sbp, DLF_MAGIC);
	marshall_LONG (sbp, DLF_GETFAC);
	q = sbp;        /* save pointer. The next field will be updated */
	msglen = 3 * LONGSIZE;
	marshall_LONG (sbp, msglen);

	/* Build request body */

	marshall_LONG (sbp, uid);
	marshall_LONG (sbp, gid);

	msglen = sbp - sendbuf;
	marshall_LONG (q, msglen);      /* update length field */


	if ((dst = (dlf_log_dst_t*)dlf_find_server(DLF_LVL_SERVICE_ONLY)) == NULL)
	  exit (SYERR);

	socket = -1;

	if (send2dlf (&socket, dst, sendbuf, msglen) < 0) {
	    exit(SYERR);
	}

	fprintf(stdout, "List of facilities:\n");
	fprintf(stdout, "===================\n");

	/* Get data */
	end_rc = 0;
	do {
	    if (getrep (socket, &reply, &rep_size, &rep_type) < 0) {
	        dlf_errmsg(NULL, DLF02, "dlflistfacilities", sstrerror(serrno));
		exit(SYERR);
	    }
	    if (rep_type == MSG_DATA) {

	      rep_buf_end = reply + rep_size;
	      rbp = reply;
	      unmarshall_BYTE(rbp, status);

	      while(rbp < rep_buf_end) {
		unmarshall_BYTE(rbp, fac_no);
		if (unmarshall_STRINGN (rbp, fac_name, DLF_MAXFACNAMELEN + 1) < 0) {
		  dlf_errmsg(NULL, DLF02, "dlflistfacilities", sstrerror(EINVAL));
		  exit(SYERR);
		}
		fprintf(stdout, "%u\t%s\n", fac_no, fac_name);
	      }
	    free(reply);
	    }
	    else if (rep_type == MSG_ERR) {
	      /* Error was printed by getrep() */
	      free(reply);
	    }
	    else if (rep_type == DLF_RC) {
	      end_rc++;
	      break;
	    }
	}
	while (status == 0);
	/* Get completion status */
	if (!end_rc) {
	    if (getrep (socket, &reply, &rep_size, &rep_type) < 0) {
	        dlf_errmsg(NULL, DLF02, "dlflistfacility", sstrerror(serrno));
		exit (SYERR);
	    }
	    else if (rep_type != DLF_RC) {
	        dlf_errmsg(NULL, DLF02, "dlflistfacility", "Protocol error");
	        exit (SYERR);
	    }
	}

#if defined(_WIN32)
	WSACleanup();
#endif
	exit (0);
}
